doc/src/Examples.dox

```diff
@@ -181,11 +181,11 @@
  * # Whenever we make new data, we always tell the manager first,
  * # This allows it to create a placeholder version or similar.
  * # NOTE: It is critical to always use the working_ref from now on.
- * working_ref = manager.preflight(entity_ref, file_spec, context)
+ * working_ref = manager.preflight([entity_ref], [file_spec], context)[0]
```

Comment:
[fix] I know we postponed deciding on how to deal with singular vs batch, and these examples show that it's not onerous on the host to deal with batch-only from a value handling perspective.

I think we are however shirking error handling a little where we just subscript the first result. The examples have never really covered error handling, but it was always via an exception, so the "explode in your face" thing is kind of pythonic. With the shift to batch, this feels to now be missing something critical.

The preceding calls never had exceptions mixed in with the results so it's less important ther. Here however, in the past, `working_ref` was always going to be a string, or you'd abort. Now, it could be an `InvalidEntityReference`, `PreflightError` or any other nasties. Given how critical the batch-first nature of the API is now, how do you feel about something along the lines of either:

Adding `throw=True` to the wrappers in `Manager`, or

```
from openassetio.hostAPI import raiseExceptions
...
working_ref = raiseExceptions(manager.preflight([entity_ref], [file_spec], context))[0]
```

Where

```
def raiseExceptions(results):
  for result in results:
    if isinstance(result, BaseException):
      raise result
  return results
```

Otherwise I'm a little concerned that we're not making the batch/error handling requirements of the host clear enough from the off. What do you think?

Reply:
Interesting.  The other alternative is to update the examples to check for errors themselves.  

Of your suggestions I lean toward a `throw=True` argument.  Though perhaps named `throwOnError` or similar - avoids having to change the name when we move to C++. It is more work for us than a `raiseExceptions` wrapper, and less explicit in the host code, but is more immediately discoverable by adopters.

Reply:
We can `[plan]` this though, lets get this PR in as-is

Reply:
Added an issue to deal with this https://github.com/TheFoundryVisionmongers/OpenAssetIO/issues/107